### PR TITLE
Add the hello-pt-resnet example

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

### Description

Add an example to show how to wrap up the Python function into a Python class for creating an object instance to use in the job API. It uses the torchvision ResNet to replace the SImpleNetwork in the hello-pt example.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/hello-world/hello-pt-resnet/src/simple_network.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
```
Comment (YuanTingHsieh) [inline examples/hello-world/hello-pt-resnet/src/simple_network.py:1]: can we rename this file to resnet18?

File: examples/hello-world/hello-pt-resnet/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+The Job API only supports the object instance created directly out of the Python Class. It does not support 
+the object instance created through using the Python function. Comparing with the hello-pt example, 
+if we replace the SimpleNetwork() object with the resnet18(num_classes=10), 
+the "resnet18(num_classes=10)" creates an torchvision "ResNet" object instance out of the "resnet18" function. The job API can 
+only use the "ResNet" object instance for generating the job config. It can not detect the object creating function logic in the "resnet18".
+
+This example demonstrates how to wrap up the resnet18 Python function into a Resnet18 Python class. Then uses the Resnet18(num_classes=10)
+object instance in the job API. After replacing the SimpleNetwork() with the Resnet18(num_classes=10),
+you can follow the exact same steps in the hello-pt example to run the fedavg_script_runner_pt.py.
```
Comment (YuanTingHsieh) [inline examples/hello-world/hello-pt-resnet/README.md:18]: We are showing how to create a network other than our SimpleNetwork in hello-pt.
To me, this is not an hello-xxx example? where should we put it? advance or tutorial?

File: examples/hello-world/hello-pt-resnet/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+Example of using [NVIDIA FLARE](https://nvflare.readthedocs.io/en/main/index.html) to train an image classifier
+using federated averaging ([FedAvg](https://arxiv.org/abs/1602.05629))
+and [PyTorch](https://pytorch.org/) as the deep learning training framework. Comparing with the Hello PyTorch example, it uses the torchvision ResNet, 
+instead of the SimpleNetwork.
+
+> **_NOTE:_** This example uses the [CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) dataset and will load its data within the client train code.
+
+The Job API only supports the object instance created directly out of the Python Class. It does not support 
+the object instance created through using the Python function. Comparing with the hello-pt example,
```
Comment (YuanTingHsieh) [inline examples/hello-world/hello-pt-resnet/README.md:11]: This section of explaining the limitations of our Job API should exist in our documentation @SYangster @chesterxgchen we need to check where to add this.

File: examples/hello-world/hello-pt-resnet/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+and [PyTorch](https://pytorch.org/) as the deep learning training framework. Comparing with the Hello PyTorch example, it uses the torchvision ResNet, 
+instead of the SimpleNetwork.
+
+> **_NOTE:_** This example uses the [CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) dataset and will load its data within the client train code.
+
+The Job API only supports the object instance created directly out of the Python Class. It does not support 
+the object instance created through using the Python function. Comparing with the hello-pt example, 
+if we replace the SimpleNetwork() object with the resnet18(num_classes=10), 
+the "resnet18(num_classes=10)" creates an torchvision "ResNet" object instance out of the "resnet18" function. The job API can 
+only use the "ResNet" object instance for generating the job config. It can not detect the object creating function logic in the "resnet18".
```
Comment (YuanTingHsieh) [inline examples/hello-world/hello-pt-resnet/README.md:14]: We should include a reference link to "resnet18" function and explain that we create the ResNet18 object similar to how function creates it.

**Review by YuanTingHsieh — COMMENTED** [review]
Great codes to show the users how to mimic the function "resnet18" to create a class that works with our Job API, suggest some adjustments

File: examples/hello-world/hello-pt-resnet/src/simple_network.py (hunk 1 above)
Comment (yhwen) [reply examples/hello-world/hello-pt-resnet/src/simple_network.py]: renamed to resnet_18.py

File: examples/hello-world/hello-pt-resnet/README.md (hunk 2 above)
Comment (yhwen) [reply examples/hello-world/hello-pt-resnet/README.md]: Added link to the torchvision resnet codes to explain.

File: examples/hello-world/hello-pt-resnet/README.md (hunk 4 above)
Comment (yhwen) [reply examples/hello-world/hello-pt-resnet/README.md]: Added

**Review by YuanTingHsieh — APPROVED** [review]
LGTM, we might also want to add this to our documentation

**Review by YuanTingHsieh — APPROVED** [review]

**Pull request merged**